From ca8f5f2d75c210fab2d7d5f4be1fc9ca17ddc10b Mon Sep 17 00:00:00 2001
From: Yelena Krivosheev <yelena@marvell.com>
Date: Tue, 15 May 2018 09:52:33 +0300
Subject: [PATCH 090/386] dts: marvell: musdk: add security acceleration module
 UIO node

Patch add Marvell security acceleration module(SAM) UIO
node to device tree for user space SAM application.

Change-Id: I98d639e3df9119290a701447be6134f9a427ee31
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/5864
Tested-by: Stefan Chulski <Stefan.Chulski@cavium.com>
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 .../devicetree/bindings/marvell-uio-sam.txt         | 13 +++++++++++++
 arch/arm64/boot/dts/marvell/armada-cp110.dtsi       |  1 +
 2 files changed, 14 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/marvell-uio-sam.txt

diff --git a/Documentation/devicetree/bindings/marvell-uio-sam.txt b/Documentation/devicetree/bindings/marvell-uio-sam.txt
new file mode 100644
index 000000000000..47a7cfb72ef0
--- /dev/null
+++ b/Documentation/devicetree/bindings/marvell-uio-sam.txt
@@ -0,0 +1,13 @@
+* Marvell UIO for Security Acceleration Module (SAM)
+
+Required properties:
+
+- compatible: should be "marvell,uio-sam".
+- eip_access: phandler to EIP crypto driver
+
+Example:
+
+CP110_LABEL(uio_sam): uio_sam@800000 {
+	compatible = "marvell,uio-sam";
+	eip_access = <&CP110_LABEL(crypto)>;
+};
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
index 5b2d9662891a..08034598947d 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
@@ -531,6 +531,7 @@
 			dma-coherent;
 			status = "disabled";
 		};
+
 		CP110_LABEL(uio_sam): uio_sam@800000 {
 			compatible = "marvell,uio-sam";
 			eip_access = <&CP110_LABEL(crypto)>;
-- 
2.17.1

